app/routes/video.rb

```diff
@@ -0,0 +1,43 @@
+class Flippd < Sinatra::Application
+  get '/videos/:id' do
+    # Save the video page view if logged in.
+    if @user then
+      VideoView.first_or_create(:video_id => params['id'].to_i, :user => @user)
+    end
+
+    @phases.each do |phase|
+      phase['topics'].each do |topic|
+        topic['videos'].each do |video|
+          if video["id"] == params['id'].to_i
```

Comment:
We could combine the contents of the loops so that there is only one loop, then an if id... else if id + 1... else if id -1...